lib/core/services/auth-guard-base.ts

```diff
@@ -70,6 +70,10 @@ export abstract class AuthGuardBase implements CanActivate, CanActivateChild {
             this.dialog.closeAll();
         }
 
+        if (this.isSilentLogin() && this.oauth2Auth && !this.oauth2Auth.isPublicUrl()) {
```

Comment:
this method has only one exit point so to make it more readable can we keep it?

Reply:
Sure but I doubt that it is more readable. I still assume that some of the SSO stuff shouldn't live in alfresco-js-api at all :-/